api/v1alpha1/vaultpkisecret_types.go

```diff
@@ -139,6 +140,10 @@ func (v *VaultPKISecret) GetIssuerAPIData() map[string]interface{} {
 		"exclude_cn_from_sans": v.Spec.ExcludeCNFromSans,
 	}
 
+	if len(v.Spec.AltNames) > 0 {
```

Comment:
It looks like strings.Join is safe to use with 0-length elements since it checks the len of the items, so this check probably isn't necessary.

Reply:
Good call!

Reply:
Fixed in https://github.com/hashicorp/vault-secrets-operator/commit/3a3be5f080719d6de65a0366ce4531b539eab877